src/components/QuizCard/index.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { useState } from "react";
+import { ADD__ANSWER } from "../../constants";
+import { useQuizData } from "../../context/quizContext";
+import { Question } from "../../types/quiz.types";
+import { AnswerButton, CardWrapper, QuestionText } from "./styles";
+
+type Props = {
+	data: Question;
+};
+
+function QuizCard({ data }: Props) {
+	const { dispatch } = useQuizData();
+	const [correct, setCorrect] = useState("");
+
+	const checkAnswer = (e: React.MouseEvent<HTMLButtonElement>) => {
+		setCorrect(e.currentTarget.value);
+		const answerObj = {
+			question: data?.question,
+			answer: e.currentTarget.value,
+			correct: e.currentTarget.value === data?.correctAnswer,
+			correctAnswer: data?.correctAnswer,
+		};
```

Comment:
Again, don't use repeated value `e.currentTarget.value` store it in a variable
Also, you can restructure `data` into
`const {question,...} = data!;`
Using underscore ensures TS that data will be available.